activity_hub.php

```diff
@@ -95,6 +95,23 @@
     $n_projects_in_state_[$project_state] = $count;
 }
 
+// SR projects have an additional qualifier beyond just the state
+// -- It can't be added to $n_projects_in_state_ because that
+// messes with the totals for the projects in PP, so create another
+// array just for SR.
+$sql = "
+    SELECT state, COUNT(*)
+    FROM projects
+    WHERE state = 'proj_post_first_checked_out' AND
+        smoothread_deadline > UNIX_TIMESTAMP()
+";
+$result = DPDatabase::query($sql);
+
+$n_projects_in_sr_ = [];
```

Comment:
Rather than create a separate array (and I understand why you need them separate) consider using a special value for the "state" key in the `$n_projects_in_state_` array.

Something like:
```php
$sql = "
    SELECT COUNT(*)
    FROM projects
    WHERE state = 'proj_post_first_checked_out' AND
        smoothread_deadline > UNIX_TIMESTAMP()
";
$result = DPDatabase::query($sql);
[$n_projects_in_state_["SR"]] = mysqli_fetch_row($result);
```

Reply:
Finally done, and thanks for the suggestion. Though it took longer than it should because I went off in the wrong direction for a while. Still, a learning experience.

I've put it in as a second commit instead of squashing, in case anyone wants to look at before and after, but I'm also happy to squash the commits into one.

Reply:
This looks good to me if you want to squash it!

Reply:
Done!